VisDataProvider.cpp

```diff
@@ -179,6 +179,17 @@ int VisDataProvider::pull() {
     return 0;
 }
 
+bool VisDataProvider::waitConnection(int s) const {
+    ALOGV("waitConnectionMs");
+    int minSleepTime = std::min(s, 1);
+    int timeSleeped = 0;
+    while (mConnectedState != STATE_CONNECTED) {
+        sleep(minSleepTime);
+        timeSleeped += minSleepTime;
+        if (timeSleeped >= s) break;
+    }
```

Comment:
As I understand, if (s=0) we will:
{
  sleep(0);
  0+=0;
  if (0>=0) break;
}
As result we will return STATE_CONNECTED instantly in any case.
Maybe it's better to implement this function as "check for connection without real sleep if (s==0)".

Reply:
With sleep time 0 it is logically correct, that client doesn't want to wait. 
In any case, mConnectedState == STATE_CONNECTED will be returned, but not STATE_CONNECTED.

Reply:
My bad. I confused '=' and '=='.
Please consider initial comment as invalid.